dev/breeze/src/airflow_breeze/utils/selective_checks.py

```diff
@@ -1767,3 +1768,125 @@ def parse_dep(dep_str: str) -> tuple[str, str | None]:
                             )
 
         return violations
+
+    @cached_property
+    def common_compat_changed_without_next_version(self) -> bool:
+        """
+        Check if common.compat provider changed and other providers changed don't have '# use next version'
+        comment for their common-compat dependency.
+        """
+        if self._github_event != GithubEvents.PULL_REQUEST:
+            return False
+
+        # First, check if common.compat provider was changed
+        common_compat_changed = False
+        for changed_file in self._files:
+            if changed_file.startswith("providers/common/compat/"):
+                common_compat_changed = True
+                break
+
+        if not common_compat_changed:
+            return False
+
+        # Find only the providers that actually have file changes (not including related providers)
+        changed_providers: set[str] = set()
+        for changed_file in self._files:
+            provider = find_provider_affected(changed_file, include_docs=False)
+            if provider and provider not in ["common.compat", "Providers"]:
+                changed_providers.add(provider)
+            elif provider == "Providers":
+                changed_providers.update(
+                    provider for provider in DEPENDENCIES.keys() if provider != "common.compat"
+                )
+                break
```

Comment:
For readability I'd reverse the check:
(1) first check for providers
(2) then elif provider and provider !=  "common-compat"